test/e2e/apps/deployment.go

```diff
@@ -1875,8 +1862,12 @@ func testPodTemplateHashCompatible(ctx context.Context, f *framework.Framework)
 	err = waitForDeploymentCompleteAndCheckRolling(c, deployment)
 	framework.ExpectNoError(err)
 	gomega.Expect(scaleResult.Spec.Replicas).To(gomega.Equal(int32(2)))
+	rs, err = testutil.GetNewReplicaSet(deployment, c)
+	framework.ExpectNoError(err, "getting rs from deployment")
+	err = e2ereplicaset.WaitForReadyReplicaSet(ctx, c, ns, rs.Name)
```

Comment:
probably no need for `WaitForReadyReplicaSet` as it should be checked by `waitForDeploymentCompleteAndCheckRolling` already

Reply:
Well, I doubt that🤔️. At first I didn't add `WaitForReadyReplicaSet` and the `e2e-gce` in ci is failing(however, the kind tests are ok). From the log it seems that rs in gce cluster is not ready. I pushed a new commit that removes the `WaitForReadyReplicaSet`, I guess we can wait for ci results to check this.

Reply:
ok, cool